OpenCV converters incompatible with OpenCV 4.0

The compiler produces the following error:
```
hipipe_core.dir/src/core/python/stream/converter.cpp.o
[ 60%] Building CXX object CMakeFiles/hipipe_core.dir/src/core/python/utility/pyboost_cv_mat_converter.cpp.o
[ 70%] Building CXX object CMakeFiles/hipipe_core.dir/src/core/python/utility/pyboost_cv_point_converter.cpp.o
/build/hipipe/src/core/python/utility/pyboost_cv_mat_converter.cpp: In member function 'cv::UMatData* hipipe::python::utility::NumpyAllocator::allocate(int, const int*, int, void*, size_t*, int, cv::UMatUsageFlags) const':                                                
/build/hipipe/src/core/python/utility/pyboost_cv_mat_converter.cpp:92:75: error: invalid conversion from 'int' to 'cv::AccessFlag' [-fpermissive]                                                                                                                             
             return stdAllocator->allocate(dims0, sizes, type, data, step, flags, usageFlags);
                                                                           ^~~~~
In file included from /usr/include/opencv4/opencv2/core.hpp:59,
                 from /usr/include/opencv4/opencv2/core/core.hpp:48,
                 from /build/hipipe/include/hipipe/core/python/utility/pyboost_cv_mat_converter.hpp:19,
                 from /build/hipipe/src/core/python/utility/pyboost_cv_mat_converter.cpp:13:
/usr/include/opencv4/opencv2/core/mat.hpp:473:69: note:   initializing argument 6 of 'virtual cv::UMatData* cv::MatAllocator::allocate(int, const int*, int, void*, size_t*, cv::AccessFlag, cv::UMatUsageFlags) const'                                                       
                                void* data, size_t* step, AccessFlag flags, UMatUsageFlags usageFlags) const = 0;
                                                          ~~~~~~~~~~~^~~~~
/build/hipipe/src/core/python/utility/pyboost_cv_mat_converter.cpp: In member function 'bool hipipe::python::utility::NumpyAllocator::allocate(cv::UMatData*, int, cv::UMatUsageFlags) const':                                                                                
/build/hipipe/src/core/python/utility/pyboost_cv_mat_converter.cpp:122:42: error: invalid conversion from 'int' to 'cv::AccessFlag' [-fpermissive]                                                                                                                            
         return stdAllocator->allocate(u, accessFlags, usageFlags);
                                          ^~~~~~~~~~~
In file included from /usr/include/opencv4/opencv2/core.hpp:59,
                 from /usr/include/opencv4/opencv2/core/core.hpp:48,
                 from /build/hipipe/include/hipipe/core/python/utility/pyboost_cv_mat_converter.hpp:19,
                 from /build/hipipe/src/core/python/utility/pyboost_cv_mat_converter.cpp:13:
/usr/include/opencv4/opencv2/core/mat.hpp:474:54: note:   initializing argument 2 of 'virtual bool cv::MatAllocator::allocate(cv::UMatData*, cv::AccessFlag, cv::UMatUsageFlags) const'                                                                                       
     virtual bool allocate(UMatData* data, AccessFlag accessflags, UMatUsageFlags usageFlags) const = 0;
                                           ~~~~~~~~~~~^~~~~~~~~~~
compilation terminated due to -fmax-errors=2.
make[2]: *** [CMakeFiles/hipipe_core.dir/build.make:128: CMakeFiles/hipipe_core.dir/src/core/python/utility/pyboost_cv_mat_converter.cpp.o] Error 1                                                                                                                           
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/hipipe_core.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```
Thank you @petrbel for reporting this.

--- a/src/core/python/utility/pyboost_cv_mat_converter.cpp
+++ b/src/core/python/utility/pyboost_cv_mat_converter.cpp
@@ -14,6 +14,12 @@
 
 namespace hipipe::python::utility {
 
+#if CV_VERSION_MAJOR == 4
+typedef cv::AccessFlag cv_access_flag_t;
+#else
+typedef int cv_access_flag_t;
+#endif
+
 //===================   ERROR HANDLING     =========================================================
 
 static PyObject* opencv_error = 0;
@@ -84,7 +90,7 @@ public:
     }
 
     cv::UMatData* allocate(int dims0, const int* sizes, int type, void* data, size_t* step,
-                           int flags, cv::UMatUsageFlags usageFlags) const
+                           cv_access_flag_t flags, cv::UMatUsageFlags usageFlags) const
     {
         if (data != 0) {
             CV_Error(cv::Error::StsAssert, "The data should normally be NULL!");
@@ -117,7 +123,7 @@ public:
         return allocate(o, dims0, sizes, type, step);
     }
 
-    bool allocate(cv::UMatData* u, int accessFlags, cv::UMatUsageFlags usageFlags) const
+    bool allocate(cv::UMatData* u, cv_access_flag_t accessFlags, cv::UMatUsageFlags usageFlags) const
     {
         return stdAllocator->allocate(u, accessFlags, usageFlags);
     }

